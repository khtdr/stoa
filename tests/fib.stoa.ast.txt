(program 
   (var a 0)
   (var temp)
   (block 
      (var b 1)
      (while (< a 5000) 
         (block 
            (block 
               (print a)
               (= temp a)
               (= a b)
            )
            (= b (+ temp b))
         )
      )
   )
   (fun fib (let [n] (block 
      (if (<= n 1) (return n))
      (return (+ (fib (- n 2)) (fib (- n 1))))
   )))
   (print (fib 20))
)
