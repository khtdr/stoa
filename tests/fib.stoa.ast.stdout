(program 
   (var a 0)
   (var temp)
   (block 
      (var b 1)
      (while (< a 5000) 
         (block 
            (block 
               (print a)
               (= temp a)
               (= a b)
            )
            (= b (+ temp b))
         )
      )
   )
   (fun fib (let [n] (block 
      (if (<= n 1) (return n))
      (return (+ (call fib (- n 2)) (call fib (- n 1))))
   )))
   (print (call fib 20))
)
